test/e2e/traceflow_test.go

```diff
@@ -0,0 +1,278 @@
+	defer func() {
+		if err := client.OpsV1alpha1().Traceflows().Delete(context.TODO(), tfIntraNode.Name, v1.DeleteOptions{}); err != nil {
+			t.Fatalf("Error when deleting traceflow: %v", err)
+		}
+	}()
+
+	client.OpsV1alpha1().Traceflows().Create(context.TODO(), tfInterNode, v1.CreateOptions{})
+	defer func() {
+		if err := client.OpsV1alpha1().Traceflows().Delete(context.TODO(), tfInterNode.Name, v1.DeleteOptions{}); err != nil {
+			t.Fatalf("Error when deleting traceflow: %v", err)
+		}
+	}()
+
+	time.Sleep(20 * time.Second)
+
+	for i := 0; i < len(testcases); i++ {
+		tc := testcases[i]
+		tf := tc.tf
+		if tf, err = client.OpsV1alpha1().Traceflows().Get(context.TODO(), tf.Name, v1.GetOptions{}); err != nil {
+			t.Errorf("Error: Get Traceflow failed: %v", err)
+		}
+		if tf.Status.Phase != tc.expectedPhase {
+			t.Errorf("Error: Traceflow Phase should be %s, but got %s", tc.expectedPhase, tf.Status.Phase)
+		}
+		if tf.Status.Results == nil || (len(tf.Status.Results) != len(tc.expectedResults)) {
+			t.Errorf("Error: Traceflow Results should be %v, but got %v", tc.expectedResults, tf.Status.Results)
+		}
+		if len(tc.expectedResults) == 1 {
+			compareObservations(tc.expectedResults[0], tf.Status.Results[0], t)
+		} else {
+			if tf.Status.Results[0].Observations[0].Component == v1alpha1.SpoofGuard {
+				compareObservations(tc.expectedResults[0], tf.Status.Results[0], t)
+				compareObservations(tc.expectedResults[1], tf.Status.Results[1], t)
+			} else {
+				compareObservations(tc.expectedResults[0], tf.Status.Results[1], t)
+				compareObservations(tc.expectedResults[1], tf.Status.Results[0], t)
+			}
+		}
+	}
+}
+
+func (data *TestData) createPod(node string, t *testing.T) string {
+	pod := randName("test-pod-")
+	t.Logf("Creating Pod '%s' on '%s'", pod, node)
+	if err := data.createBusyboxPodOnNode(pod, node); err != nil {
+		t.Fatalf("Error when creating Pod '%s': %v", pod, err)
+	}
+	if _, err := data.podWaitForIP(defaultTimeout, pod, testNamespace); err != nil {
+		t.Errorf("Error when waiting for IP for Pod '%s': %v", pod, err)
+	}
+	return pod
+}
+
+func fillTraceflow(src string, dest string) *v1alpha1.Traceflow {
+	return &v1alpha1.Traceflow{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: randName("test-traceflow-"),
+		},
+		Spec: v1alpha1.TraceflowSpec{
+			Source: v1alpha1.Source{
+				Namespace: testNamespace,
+				Pod:       src,
+			},
+			Destination: v1alpha1.Destination{
+				Namespace: testNamespace,
+				Pod:       dest,
+			},
+			Packet: v1alpha1.Packet{
+				IPHeader: v1alpha1.IPHeader{
+					Protocol: 6,
+				},
+				TransportHeader: v1alpha1.TransportHeader{
+					TCP: &v1alpha1.TCPHeader{
+						DstPort: 80,
+					},
+				},
+			},
+		},
+	}
+}
+
+func compareObservations(expected v1alpha1.NodeResult, actual v1alpha1.NodeResult, t *testing.T) {
+	if expected.Node != actual.Node {
+		t.Errorf("Error: NodeResult should be on %s, but now on %s", expected.Node, actual.Node)
```

Comment:
s/but now on/but is on

Reply:
Updated.

Working on your suggestion on kind.